### ISSUE TITLE
Cumulative Sums and Counts not working with MySQL 5.7 in v0.50 due to CTE 

### ISSUE THREAD

**Issue by tomeli5n** [metabase/metabase#45142]
### Describe the bug

Starting from version v0.50 of Metabase, it is not possible to perform cumulative sums or cumulative counts using MySQL 5.7.

This functionality was working in v0.49

I can not found in changelog if 5.7 was deprecated, just this issue #26328

### To Reproduce

1. Create a simple question with comulative sum and mysql 5.7
2. See error
 ```
You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(ORDER BY `source`.`FechPago` ASC ROWS UNBOUNDED PRECEDING) AS `count` FROM (SEL' at line 2
```

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "es-ES",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.23+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.23",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.23+9",
    "os.name": "Linux",
    "os.version": "3.10.0-1062.1.2.el7.x86_64",
    "user.language": "en",
    "user.timezone": "America/Argentina/Buenos_Aires"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "mysql",
      "sqlserver",
      "sqlite",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MariaDB",
        "version": "10.3.17-MariaDB-1:10.3.17+maria~bionic"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.7.10"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-07-02",
      "tag": "v0.50.9",
      "hash": "720e549"
    },
    "settings": {
      "report-timezone": "America/Argentina/Buenos_Aires"
    }
  }
}
```

### Severity

blocking and upgrade

### Additional context

v0.49 sql:

```sql
SELECT
  `pago`.`FechPago` AS `FechPago`,
  SUM(`pago`.`TotaPago`) AS `sum`
FROM
  `pago`
WHERE
  (`pago`.`PeriPago` = 2024)
 
   AND (
    `pago`.`FechPago` >= STR_TO_DATE(
      CONCAT(DATE_FORMAT(NOW(6), '%Y-%m'), '-01'),
      '%Y-%m-%d'
    )
  )
  AND (
    `pago`.`FechPago` < STR_TO_DATE(
      CONCAT(
        DATE_FORMAT(DATE_ADD(NOW(6), INTERVAL 1 month), '%Y-%m'),
        '-01'
      ),
      '%Y-%m-%d'
    )
  )
GROUP BY
  `pago`.`FechPago`
ORDER BY
  `pago`.`FechPago` ASC
```
v0.50 sql with `OVER ... AS` statement

```sql
SELECT
  `source`.`FechPago` AS `FechPago`,
  SUM(COUNT(*)) OVER (
   
ORDER BY
      `source`.`FechPago` ASC ROWS UNBOUNDED PRECEDING
  ) AS `count`
FROM
  (
    SELECT
      DAYOFMONTH(`pago`.`FechPago`) AS `FechPago`
    FROM
      `pago`
   
WHERE
      (`pago`.`PeriPago` = 2024)
     
   AND (
        `pago`.`FechPago` >= STR_TO_DATE(
          CONCAT(DATE_FORMAT(NOW(6), '%Y-%m'), '-01'),
          '%Y-%m-%d'
        )
      )
      AND (
        `pago`.`FechPago` < STR_TO_DATE(
          CONCAT(
            DATE_FORMAT(DATE_ADD(NOW(6), INTERVAL 1 month), '%Y-%m'),
            '-01'
          ),
          '%Y-%m-%d'
        )
      )
  ) AS `source`
GROUP BY
  `source`.`FechPago`
ORDER BY
  `source`.`FechPago` ASC
```

**Comment by camsaul** [comment]
Hi, we don't officially support MySQL 5.7, it has been unsupported upstream for 8 months. https://endoflife.date/mysql

**Comment by ngm** [comment]
I encountered this, and just to report for future travellers that those questions failing when pointing at MySQL 5.7 do work fine when upgrading MySQL to v8.
